packages/path_provider/path_provider/lib/path_provider.dart

```diff
@@ -3,14 +3,36 @@
 // found in the LICENSE file.
 
 import 'dart:async';
-import 'dart:io' show Directory;
+import 'dart:io' show Directory, Platform;
 
+import 'package:flutter/foundation.dart' show visibleForTesting;
+import 'package:path_provider_linux/path_provider_linux.dart';
 import 'package:path_provider_platform_interface/path_provider_platform_interface.dart';
 
 export 'package:path_provider_platform_interface/path_provider_platform_interface.dart'
     show StorageDirectory;
 
-PathProviderPlatform get _platform => PathProviderPlatform.instance;
+// Should not be used in users' code, just made visible for testing to test the method channel implementation
+@visibleForTesting
+set testOverrideUseMethodChannel(bool override) {
+  _useMethodChannel = override;
+}
+
+bool _useMethodChannel = false;
+PathProviderPlatform __platform;
+
+// This is to manually endorse the linux path provider until automatic registration of dart plugins is implemented.
+PathProviderPlatform get _platform {
+  if (__platform != null) {
+    return __platform;
+  }
+  if (Platform.isLinux && !_useMethodChannel) {
```

Comment:
Do we need a `kIsWeb` check first? I'm not sure what `Platform.isLinux` returns when running Flutter for Web on Linux.